src/components/earn/ImportFarmModal.tsx

```diff
@@ -0,0 +1,162 @@
+import { getAddress } from '@ethersproject/address'
+import { ButtonError } from 'components/Button'
+import { LightCard } from 'components/Card'
+import { SearchInput } from 'components/SearchModal/styleds'
+import { useCustomStakingInfo } from 'pages/Earn/useCustomStakingInfo'
+import { FarmSummary } from 'pages/Earn/useFarmRegistry'
+import React, { RefObject, useContext, useEffect, useRef, useState } from 'react'
+import { Text } from 'rebass'
+import styled, { ThemeContext } from 'styled-components'
+import { isAddress } from 'web3-utils'
+
+import { BIG_INT_SECONDS_IN_WEEK } from '../../constants'
+import { CloseIcon, TYPE } from '../../theme'
+import { AutoColumn } from '../Column'
+import Modal from '../Modal'
+import Row, { RowBetween } from '../Row'
+
+const ContentWrapper = styled(AutoColumn)`
+  width: 100%;
+  flex: 1 1;
+  position: relative;
+  padding: 1rem;
+`
+
+interface ImportFarmModalProps {
+  isOpen: boolean
+  onDismiss: () => void
+  farmSummaries: FarmSummary[]
+}
+
+export default function ImportFarmModal({ isOpen, onDismiss, farmSummaries }: ImportFarmModalProps) {
+  const inputRef = useRef<HTMLInputElement>()
+  const theme = useContext(ThemeContext)
+  const [farmAddress, setFarmAddress] = useState<string>('')
+  const { stakingToken, rewardTokens, totalRewardRates, valueOfTotalStakedAmountInCUSD } =
+    useCustomStakingInfo(farmAddress)
+
+  const [error, setError] = useState<string | undefined>(undefined)
+
+  const farmExists = isAddress(farmAddress)
+    ? farmSummaries.find(
+        (farmSummary) => farmAddress && getAddress(farmSummary.stakingAddress) === getAddress(farmAddress)
+      )
+    : undefined
+
+  const importedFarms = localStorage.getItem('imported_farms')
+
+  const handleInput = (event: any) => {
+    const input = event.target.value
+    setFarmAddress(input)
+  }
+
+  useEffect(() => {
+    if (isAddress(farmAddress)) {
+      const res = importedFarms
+        ? [...JSON.parse(importedFarms)].find((item) => getAddress(item) === getAddress(farmAddress))
+        : undefined
+      setError(res ? 'The farm has already been imported' : undefined)
+    } else {
+      if (farmAddress.length > 0) {
+        setError('Enter valid farm address')
+      } else {
+        setError(undefined)
+      }
+    }
+  }, [farmAddress, importedFarms])
+
+  const onConfirm = () => {
+    localStorage.setItem(
+      'imported_farms',
+      JSON.stringify(importedFarms ? [...JSON.parse(importedFarms), farmAddress] : [farmAddress])
```

Comment:
What happens if `importedFarms` is ever a malformed JSON?

E.g. try `JSON.parse('af8= 9a8')`

We should catch instances of this where we could potentially crash the app and gracefully reset the local storage if we detect that this happens.

Reply:
Also, what happens if any of the farms in `importedFarms` is maliciously updated to not be a farm? Do we handle this case gracefully too?